todoCLI/todoCLI.js

```diff
@@ -0,0 +1,153 @@
+const rl = require(`readline`);
+const fs = require(`fs`);
+
+const menu= `(V) View ✏︎ (n) New ✏︎ (cX) Complete  ✏︎  (dX) Delete ✏︎ (q) Quit\n`;
+
+console.log(`\n\nWelcome to Todo CLI!\n----------------------`);
+const interface = rl.createInterface ( {
+	input: process.stdin,
+	output: process.stdout
+});
+
+waitingForUser();
+
+function waitingForUser () {
+  interface.question(menu, (command) => {
+    let number = parseFloat(command.replace(/[cd]/ , ''))
+    command = command.replace(number, '')
```

Comment:
This might be running before you're walking, but in case you're interested, a "fancy" way of doing this with regex  (using "capture groups") would be:
```javascript 
let match = command.match(/([vncdq])(\d*)/i)
let letter = match[1];
let number = match[2];
```

Reply:
is this instead of the switch?

Reply:
I would suggest check for valid commands up front and use this function as sort of a router to other actions.
First thing would be to set up some ["guard causes"](https://medium.com/@scadge/if-statements-design-guard-clauses-might-be-all-you-need-67219a1a981a) against bad input! 😠 

I'm thinking something like:

```javascript
function waitingForUser() {
  interface.question(menu, (userInput) => {
    if (userInput) {
      waitForUser(); // missing input. Start again.
    }
    if (userInput && userInput.length > 2) {
      waitForUser(); // too many characters! Start again.
    }
    
    // now we know there is a value and it is NOT bigger than 2.
    // we'll check for what the first character is to determine what to do next.
    const userInput = userInput.toLowerCase();
    const acceptableCommands = ["c", "d", "n", "v", "q"];
    const first = command[0];
    
    // we can use `find` to see if the first character exists in the acceptable list and it
    // will return it! You can use regex here as well if that makes more sense to you.
    const command = acceptableCommands.find((command) => command === first);
    
    // if `command` exists then we know it's a good command
    // now we can check to see which command it is do something with it
    if (command === "c") {
      // so now, let's see what the other value is. 
      const taskNumber = parseInt(command[1]);
      // next we'll need to see if this is a valid task number.
      // we'll do that by checking to see if the number within the range of the list array
      if (taskNumber !== NaN && (taskNumber >= 0 && task < todoList.length)) {
        // ok great, so it's a number and it in the range of our list
        completeTask(taskNumber);
      } else {
        // send them off the a new function which you can recursively repeat until they
        // give you a number you like.
      }
    }

    if (command === "d") {
      // so now, let's see what the other value is. 
      const taskNumber = parseInt(command[1]);
      // next we'll need to see if this is a valid task number.
      // we'll do that by checking to see if the number within the range of the list array
      if (taskNumber !== NaN && (taskNumber >= 0 && task < todoList.length)) {
        // ok great, so it's a number and it in the range of our list
        deleteTask(taskNumber);
      } else {
        // send them off the a new function which you can recursively repeat until they
        // give you a number you like.
      }
    }

    // these following command don't require a second value so we'll just ignore it.
    if (command === "n") {
      addTask();
    }

    if (command === "v") {
      viewToDoList();
    }

    if (command === "q") {
      process.exit();
    }
  });
};
```
Let me know if you want to go through it together 😄